stdlib/src/builtin/bool.mojo

```diff
@@ -322,6 +327,15 @@ struct Bool(
         """
         return __mlir_op.`index.casts`[_type = __mlir_type.index](self.value)
 
+    @always_inline("nodebug")
```

Comment:
Please add a test for this (we test these implementations directly using explicit calls to `__index__` instead of through the free functions).